.github/workflows/build-plugin-archive.yml

```diff
@@ -0,0 +1,219 @@
+
+      #
+      # PHP dependencies
+      #
+
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+
+      - name: Install Composer dependencies with dev dependencies
+        # ⬑ The idea is to ensure we have non-production tools like composer-assets-compiler and translation downloader available
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: ${{ inputs.COMPOSER_ARGS_DEV }}
+
+      - name: Check optional composer build tools
+        id: composer-tools
+        run: |
+          hasDep(){
+            composer show -i -N -D --strict "${1}" >/dev/null 2>&1
+            local EXIT=$?
+            echo "$EXIT"
+          }
+          echo "assets-compiler=$( hasDep inpsyde/composer-assets-compiler )" >> $GITHUB_OUTPUT
+          echo "translation-downloader=$( hasDep inpsyde/wp-translation-downloader )" >> $GITHUB_OUTPUT
+
+      - name: Run Composer Assets Compiler
+        if: steps.composer-tools.outputs.assets-compiler == '0'
+        run: composer compile-assets
+
+      - name: Run WP Translation Downloader
+        if: steps.composer-tools.outputs.translation-downloader == '0'
+        run: composer wp-translation-downloader:download
+
+      - name: Install Composer dependencies without dev dependencies
+        # ⬑ Now we want to get rid of dev dependencies and most probably skip composer scripts
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: ${{ inputs.COMPOSER_ARGS }}
+
+
+      #
+      # JS environment. Only for setting up node/npm versions. Actual tasks should be run at project level
+      #
+
+      - name: Check for package.json
+        id: check-for-package
+        run: |
+          if [ -f package.json ]; then
+            echo "has_package=true" >> $GITHUB_OUTPUT
+          else
+            echo "has_package=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Set up node
+        if: steps.check-for-package.outputs.has_package == 'true'
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+          registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
+
+      - name: Cache npm dependencies
+        if: steps.check-for-package.outputs.has_package == 'true'
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      #
+      # Package metadata
+      #
+
+      - name: Add commit hash to plugin header
+        env:
+          GIT_SHA: ${{ steps.git.outputs.sha_short }}
+        run: 'sed -Ei "s/SHA: .*/SHA: ${GIT_SHA}/g" ${{ inputs.PLUGIN_MAIN_FILE }}'
+
+      - name: Set plugin version header
+        env:
+          PACKAGE_VERSION: ${{ github.event.inputs.packageVersion }}
+        run: 'sed -Ei "s/Version: .*/Version: ${PACKAGE_VERSION}/g" ${{ inputs.PLUGIN_MAIN_FILE }}'
```

Comment:
Can we do it like here? Optional and adds the current version if missing in the input (that is if you enter `somebugfix` it becomes `1.2.3-somebugfix`).

https://github.com/woocommerce/woocommerce-paypal-payments/blob/664a53dd36ed70b68745bd21552b0160d7b58462/.github/workflows/package.yml#L32-L38

Reply:
We discussed this and agreed that the nature of the version string is best defined and maintained by the plugin developers. The reusable workflow would just take the input and use it as-is without attaching any logic or meaning to it.

I added a note about this to the documentation